src/main/java/com/github/manolo8/darkbot/gui/players/PlayerEditor.java

```diff
@@ -115,6 +115,7 @@ public void onEntityPlayerAdd(EntitiesAPI.EntityCreateEvent event) {
                     PlayerInfo info = playersModel.get(idx);
                     nearbyPlayers.add(info);
                 }
+                refreshList();
```

Comment:
if you call refreshList here, there's no longer a point to calling addElement at all, since refreshList would already re-create the whole thing.

The point is you cannot do refreshList because that method does not calculate the difference (remove the uneeded, and add the new ones) instead it wipes the whole list and re-creates it, meaning any selection is lost.

One of my original points was you could rewrite refreshList to be able to smartly add/remove things from the list instead of wiping it and re-creating, and that would simplify the rest of your code as you would only need to add or remove from nearbyPlayers and call refreshList, without bothering about any other adding/removing logic and avoiding duplicates